src/basic/uid-classification.c

```diff
@@ -109,23 +110,17 @@ const UGIDAllocationRange *acquire_ugid_allocation_range(void) {
         return &default_ugid_allocation_range;
 }
 
-bool uid_is_system(uid_t uid) {
-        const UGIDAllocationRange *defs;
-        assert_se(defs = acquire_ugid_allocation_range());
+bool uid_is_dynamic(uid_t uid) {
+        char path[STRLEN("/run/systemd/dynamic-uid/") + DECIMAL_STR_MAX(uid_t) + 1];
```

Comment:
sorry, but this is a quite a different function now. It suddenly answers the question of "is this currently an allocated dynamic uid". Instead of "this is possibly ever going to be used as a dynamic uid or was it possibly?".

Also, it now assumes we universally have access to an fs dir, which is quite a difference too.